### ISSUE TITLE
Sometimes, I can't send or receive messages after socket is disconnected until get specific log.

### ISSUE THREAD

**Issue by seubseub** [GetStream/stream-chat-swift#953]
## What did you do?
I am using self-made chatVC,
Send and receive simple messages.

## What did you expect to happen?
I want to use it without problem for sending and receiving messages.

## What happened instead?
At first, the two users send and receive messages to some extent.

You will receive a log like the one below.
Sending/receiving messages does not work after receiving this log.
``` 
[tcp] tcp_input [C10.1:3] flags=[R] seq=559879131, ack=0, win=0 state=CLOSED
```
After getting an error like this,
```
Task <FE2E3CE8-6C0E-4BEB-933A-9AD2E3916637>.<1> finished with error [57] Error Domain=NSPOSIXErrorDomain Code=57 "Socket is not connected" UserInfo={NSErrorFailingURLStringKey=wss://chat-proxy-dublin. stream-io-api.com/connect?api_key=c67dgueh4s94&json=%7B%22user_details%22:%7B%22id%22:%222CTR9PSJnZP%22%7D,%22user_id%22:%222CTR9PSJnZP%22,%22server_determines_connection_id22 true%7D, NSErrorFailingURLKey=wss://chat-proxy-dublin.stream-io-api.com/connect?api_key=c67dgueh4s94&json=%7B%22user_details%22:%7B%22id%22:%222CTR9PSJnZP%22%7D ,%22user_id%22:%222CTR9PSJnZP%22,%22server_determines_connection_id%22:true%7D, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalWebSocketTask <FE2E3CE8-6C0E-4BEB-933A-9AD2E3916637>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalWebSocketTask <FE2E3CE8-6C0E-4BEB-933A-9AD2E3916637>.<1>}
```

Sending/receiving messages again works well after receiving an error log stating that the socket has been disconnected.

Whether the above log is an error related to this streamChat,
I'm also using libraries called'PusherSwift' and'~> 9.0', so I'm curious if the two libraries conflict.

## GetStream Environment
**GetStream Chat version 3.1.1:**
**GetStream Chat frameworks:** StreamChat 3.1.1, StreamChatClient 3.1.1, StreamChatCore 3.1.1
**iOS version: 14.4.2**
**Swift version: 5.3?**
**Xcode version: 12.4**
**Device: iphone 12 Pro, iphone 8**

## Additional context
In the case of using StreamChatUI,
Even though the above socket is closed, sending and receiving messages are working well. Can anybody know what's wrong?
https://github.com/GetStream/stream-chat-swift/wiki/Cheat-Sheet
I also refer to this cheat-sheet a lot.

If you are aware of similar issues or causes, please let us know.
Thank you. :)

**Comment by seubseub** [comment]
I think after call synchronize,
remote data fetched is too late.
what should I do??
thx :)

**Comment by cf-lam** [comment]
I'm experiencing the same issue. Here's my error line:

2021-04-15 03:47:18.895540-0500 CommunityApp[53814:17051504] Task <1BF4BA60-94DC-4799-9C5B-A5EA3933A863>.<1> finished with error [57] Error Domain=NSPOSIXErrorDomain Code=57 "Socket is not connected" UserInfo={NSErrorFailingURLStringKey=wss://chat-proxy-dublin.stream-io-api.com/connect?api_key=<redacted>&json=%7B%22user_details%22:%7B%22id%22:%2221205%22%7D,%22server_determines_connection_id%22:true,%22user_id%22:%2221205%22%7D, NSErrorFailingURLKey=wss://chat-proxy-dublin.stream-io-api.com/connect?api_key=<redacted>&json=%7B%22user_details%22:%7B%22id%22:%2221205%22%7D,%22server_determines_connection_id%22:true,%22user_id%22:%2221205%22%7D, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalWebSocketTask <1BF4BA60-94DC-4799-9C5B-A5EA3933A863>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalWebSocketTask <1BF4BA60-94DC-4799-9C5B-A5EA3933A863>.<1>}

**Comment by VojtaStavik** [comment]
Hi @seubseub, we fixed an issue that might be related to the problem you experienced. Can you please try to use the version of the SDK from this commit b6b9c44c6bac2de35e0908d7d0ba044d6fa282c9 and let me know If you still experience the issue?

**Comment by cf-lam** [comment]
I'm experiencing the same issue, I just tried that commit @VojtaStavik 

Here's my error log:

2021-04-15 12:02:20.980866-0500 CommunityApp[61138:17443853] Connection 3: missing error, so heuristics synthesized error(1:53)
2021-04-15 12:02:20.981053-0500 CommunityApp[61138:17443853] Connection 3: encountered error(1:53)
2021-04-15 12:02:20.987005-0500 CommunityApp[61138:17448707] Task <17095AA5-C681-4505-8BA1-9D23DBC47D9D>.<1> finished with error [57] Error Domain=NSPOSIXErrorDomain Code=57 "Socket is not connected" UserInfo={NSErrorFailingURLStringKey=wss://chat-proxy-dublin.stream-io-api.com/connect?api_key=<redacted>&json=%7B%22server_determines_connection_id%22:true,%22user_id%22:%2221431%22,%22user_details%22:%7B%22id%22:%2221431%22%7D%7D, NSErrorFailingURLKey=wss://chat-proxy-dublin.stream-io-api.com/connect?api_key=<redacted>&json=%7B%22server_determines_connection_id%22:true,%22user_id%22:%2221431%22,%22user_details%22:%7B%22id%22:%2221431%22%7D%7D, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalWebSocketTask <17095AA5-C681-4505-8BA1-9D23DBC47D9D>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalWebSocketTask <17095AA5-C681-4505-8BA1-9D23DBC47D9D>.<1>}
2021-04-15 12:02:20.981 [ERROR] [io.getStream.chat.core.web_socket_engine_queue] [URLSessionWebSocketEngine.swift:62] [doRead()] > Failed receiving Web Socket Message with error: Error Domain=NSPOSIXErrorDomain Code=57 "Socket is not connected" UserInfo={NSErrorFailingURLStringKey=wss://chat-proxy-dublin.stream-io-api.com/connect?api_key=<redacted>&json=%7B%22server_determines_connection_id%22:true,%22user_id%22:%2221431%22,%22user_details%22:%7B%22id%22:%2221431%22%7D%7D, NSErrorFailingURLKey=wss://chat-proxy-dublin.stream-io-api.com/connect?api_key=<redacted>&json=%7B%22server_determines_connection_id%22:true,%22user_id%22:%2221431%22,%22user_details%22:%7B%22id%22:%2221431%22%7D%7D}

**Comment by VojtaStavik** [comment]
Hi @seubseub, I believe this is fixed in `v3.1.8` and `v3.2.0-beta.8`. Can you please try and re-open the issue in case you're experiencing the problem even after upgrading?
